src/components/Select/Select.tsx

```diff
@@ -0,0 +1,130 @@
+import { SelectProps } from './Select.types';
+import React, { useState, useMemo } from 'react';
+import {
+  FormControl,
+  Select as MuiSelect,
+  MenuItem,
+  InputLabel,
+  ListSubheader,
+  TextField,
+  InputAdornment,
+  Divider,
+  Box,
+  CircularProgress,
+} from '@mui/material';
+import SearchIcon from '@mui/icons-material/Search';
+
+/**
+ * This component kets the user select an item from a dropdown.
+ *
+ * @return {JSX.Element}
+ */
+const Select: React.FC<SelectProps> = ({
+  options,
+  option,
+  fixedOptions,
+  setOption,
+  loading,
+}) => {
+  const containsText = (text: string, searchText: string) =>
```

Comment:
As a suggestion, you could move this function outside of component. Because you are not using state for it and as example when component re-renders it goes through this function as well, because it is inside of component. So it is just a performance/cleanliness thing.

Reply:
Done!